x/auth/signing/verify.go

```diff
@@ -57,6 +59,25 @@ func internalSignModeToAPI(mode signing.SignMode) (signingv1beta1.SignMode, erro
 	}
 }
 
+// VerifysignatureCache is a cache of verified signatures
+// If the cached item is not found, it will be verified and added to the cache
+func verifySig(signBytes, sig []byte, pubKey cryptotypes.PubKey) bool {
+	bz := sha256.Sum256(append(signBytes, sig...))
+	hash := bz[:]
+	cachePub, ok := SignatureCache().Get(hash)
```

Comment:
Not sure why were hashing the signature, why not use the raw representation or (string) of it directly?